Mathlib/Algebra/Group/Prod.lean

```diff
@@ -37,8 +37,8 @@ variable {A : Type _} {B : Type _} {G : Type _} {H : Type _} {M : Type _} {N : T
 
 namespace Prod
 
-@[to_additive]
-instance [Mul M] [Mul N] : Mul (M × N) :=
+@[to_additive instAddProd]
+instance instMulProd [Mul M] [Mul N] : Mul (M × N) :=
```

Comment:
Ideally these should be called `Prod.instAdd`  not `Prod.instAddProd`. Was this a deliberate choice, or just an artifact of some automation?

Reply:
We're inconsistent this (i.e. does being in the namespace mean you leave off that name as a terminal fragment of the an instance name?) across Mathlib. This PR is not attempting to resolve the question of "how does Mathlib want to name things" or to match names to a not-yet-existing consensus about that question. It just ensures we never rely on a Lean 4 auto-generated name, to ensure that it is possible to make changes in Lean 4 without having to fight Mathlib.

Nevertheless, I'll change this file now.